src/interrupts/src/lib.rs

```diff
@@ -168,32 +173,24 @@ pub extern "C" fn interrupt_handler(interrupt_number: isize, error_code: isize)
         33 => keyboard_handler(),
         _ => panic!("interrupt {} with error code {:x}", interrupt_number, error_code),
     }
-    unsafe{
-        match interrupt_number {
-            i if i >= 40 => {
-                outb(0xA0, 0x20);
-                outb(0x20, 0x20);
-            },
-            32...40 => outb(0x20, 0x20),
-            _ => {},
-        }
-        enable();
-    };
+
+    pic::eoi_for(interrupt_number);
+
+    enable();
 }
 
 fn keyboard_handler() {
     let scancode = unsafe { inb(0x60) };
-    kprintln!("code: {}", scancode);
+
+    keyboard::getchar(scancode).map(|c| {
+        keyboard::write_to_buffer(c);
+        kprint!("{}", c as char);
+    });
 }
 
 #[inline]
-pub unsafe fn inb(port: u16) -> u8 {
+unsafe fn inb(port: u16) -> u8 {
```

Comment:
same function defined at src/pic/src/lib.rs..
this crate can also be used https://github.com/gz/rust-x86..